src/Microsoft.Framework.PackageManager/Restore/NuGet/HttpSource.cs

```diff
@@ -129,7 +129,7 @@ internal class HttpSource
             {
                 HttpStatusCode statusCode;
 
-                using (var response = await _client.SendAsync(request))
+                using (var response = await _client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead))
                 {
                     response.EnsureSuccessStatusCode();
```

Comment:
The FileStream using extends to the end of this method. Extending the response using around all of that would make your nesting even deeper. Just a style issue.

But honestly there's not much point in optimizing the failure case. Perf primarily matters in the success case.